lib/types/number.js

```diff
@@ -37,27 +37,6 @@ module.exports = Any.extend({
                 result.value = 0;           // -0
             }
 
-            if (!schema._flags.unsafe) {
```

Comment:
Probably, this is not what is needed to do in order to solve the problem... But I'm really wondering what is the responsability of this block...

Reply:
This code converts back the parsed number into exponential notation to check whether there was a loss during parsing. Unless you have a better way to check for precision loss, this must stay.